include/tvm/relax/transform.h

```diff
@@ -322,11 +322,20 @@ class FusionPatternNode : public Object {
    */
   Optional<PackedFunc> check;
 
+  /*!
+   * \brief The function to get attributes for fused function
+   *
+   * It should have signature
+   * Map<String, String>(const PatternCheckContext& context)
```

Comment:
At first glance I don't understand why it needs to take `PatternCheckContext` as argument.

Reply:
A context is needed to set extra attributes. The context should have access to all expressions in the pattern, I find out that PatternCheckContext is a good one. It contains all the expressions marked during pattern matching. Some attributes need to be defined based on the expressions

Reply:
What `CreatePatternCheckContext` does is very simple. I suggest extracting what you need from there into a function, and use it in both here and inside `CreatePatternCheckContext`.

Reply:
@Archermmt Please update the function signature above.